gpu-detection/src/lib.rs

```diff
@@ -0,0 +1,122 @@
+use anyhow::{bail, Context};
+use model::{Clocks, Cuda, Gpu, Memory};
+use nvml_wrapper::{enum_wrappers::device::Clock, Device, Nvml};
+
+pub mod model;
+
+pub struct GpuDetection {
+    nvml: Nvml,
+}
+
+impl GpuDetection {
+    pub fn init() -> anyhow::Result<Self> {
+        let nvml = Nvml::init()?;
+        Ok(Self { nvml })
+    }
```

Comment:
But it's not so important problem for now

Reply:
Function `::init` dynamically loads `nvml`. I call it directly because it makes `ya-runtime-ai test` command result more transparent than having it failed in some static initializer.

Reply:
I suspect that some kind of lazy initialization that gives you an error when you need it would be possible, but as I said let's leave this as is